app/assets/src/api/upload.js

```diff
@@ -124,6 +125,7 @@ export const bulkUploadLocal = ({
       .then(response => {
         // After successful sample creation, upload to the presigned URLs returned
         const sampleId = response.id;
+        startUploadHeartbeat(sampleId);
```

Comment:
Can the client report some kind of progress metric?

Reply:
@tfrcarvalho I think we could break it down by input file and do an upload progress per file, but I think it'd be inelegant with the current "3 pages" (old single, old bulk, new consolidated) so I'd prefer to refactor it when we switch over to the new flow? I think the state would need a structure of sampleIds => files => %progress which gets updated onUploadProgress

Reply:
OK, i think it will be good to have that info for multiple reasons, but can definitively be a follow up.